backend/typescript/services/implementations/reviewService.ts

```diff
@@ -498,18 +499,41 @@ class ReviewService implements IReviewService {
     return result;
   }
 
-  async getReviews(): Promise<ReviewResponseDTO[]> {
-    let reviews: PgReview[];
-    let result: ReviewResponseDTO[];
+  getPaginationOffset(page: string, limit: number | undefined): number {
+    const offset = page && limit ? parseInt(page, 10) * limit : 0;
+    return offset;
+  }
+
+  async getReviews(
+    page: string,
+    size: string,
+  ): Promise<PaginatedReviewResponseDTO> {
+    let result: PaginatedReviewResponseDTO;
 
     try {
       result = await this.db.transaction(async (t) => {
-        reviews = await PgReview.findAll({
+        const limit = size ? parseInt(size, 10) : undefined;
+        const offset = this.getPaginationOffset(page, limit);
+
+        const { rows, count } = await PgReview.findAndCountAll({
           transaction: t,
           include: [{ all: true, nested: true }],
+          limit,
```

Comment:
@chriistine I tried adding a `distinct: true` field here, and it seemed to solve the incorrect count issue. This may have other side effects tho, as it could be filtering distinct associations (ie tags/genres might show up only for 1 book even if they've been assigned to multiple books), so I'd suggest thoroughly testing if we're going that route :thinking:

Reply:
For reference: https://github.com/sequelize/sequelize/issues/10557